packages/state-transition/src/block/processWithdrawals.ts

```diff
@@ -0,0 +1,68 @@
+import {ssz, capella} from "@lodestar/types";
+import {MAX_EFFECTIVE_BALANCE, ETH1_ADDRESS_WITHDRAWAL_PREFIX, MAX_WITHDRAWALS_PER_PAYLOAD} from "@lodestar/params";
+
+import {CachedBeaconStateCapella} from "../types.js";
+import {decreaseBalance} from "../util/index.js";
+
+export function getExpectedWithdrawals(state: CachedBeaconStateCapella): capella.Withdrawal[] {
+  const currentEpoch = state.epochCtx.epoch + 1;
+  let withdrawalIndex = state.nextWithdrawalIndex;
+  let validatorIndex = state.nextWithdrawalValidatorIndex;
+  const {validators, balances} = state;
+
+  const withdrawals: capella.Withdrawal[] = [];
+  for (let index = 0; index < validators.length; index++) {
```

Comment:
is this `index` variable used inside the for loop? Looks like we use `validatorIndex` only

Reply:
yes, could replace index by i so as to provide clarity that this is just a bounded loop